### ISSUE TITLE
coordsystem.json files should not contain task field

### ISSUE THREAD

**Issue by jasmainak** [bids-standard/legacy-validator#848]
@robertoostenveld caught this issue in https://github.com/bids-standard/bids-examples/issues/167, however I think we should make a more permanent fix to this problem by fixing the validator to catch such files.

**Comment by sappelhoff** [comment]
@jasmainak could you be a bit more elaborate what needs to be done? That'd make it easier for somebody else to pick up the problem (unless you want to fix it yourself and this is only your reminder)

**Comment by jasmainak** [comment]
The details are in the reference issue but let me copy-paste to clarify:

```
sub-<label>/
    [ses-<label>]/
      eeg/
        [sub-<label>[_ses-<label>][_acq-<label>]_coordsystem.json]
```

does not have `task-<xxx>` in it, as mentioned in the specification. But the validator does not raise an error if it has the task field in there

**Comment by sappelhoff** [comment]
I see, thanks - that'd have to be changed in the regular expressions here: https://github.com/bids-standard/bids-validator/blob/0f55b8f4b2f6e608a8a3cb0aee93fa732e898307/bids-validator/bids_validator/rules/file_level_rules.json#L159-L220

it could mean that we have to change the structure a bit ... because solving this with a regular expression would make that expression even more complicated and bloated than it already is :grimacing:

**Comment by effigies** [comment]
Edit: sorry, misread this as about the json... Ignore me.

**Comment by jasmainak** [comment]
hmm ... as long as old and new tests pass, we are fine :)

**Comment by effigies** [comment]
We have permitted more entities for coordsystem.json to avoid massive breakage of datasets. Backwards-incompatible changes to the spec can remove them again, and that will be resolved in the schema.
